api/healthcheck.go

```diff
@@ -10,32 +10,53 @@
 package api
 
 import (
+	"context"
 	"fmt"
 	"net/http"
 	"strings"
 
-	"github.com/labstack/echo"
+	"go.uber.org/zap"
+
+	api "github.com/topfreegames/podium/proto/podium/api/v1"
 )
 
-// HealthCheckHandler is the handler responsible for validating that the app is still up
-func HealthCheckHandler(app *App) func(c echo.Context) error {
-	return func(c echo.Context) error {
-		c.Set("route", "Healthcheck")
-		workingString := app.Config.GetString("healthcheck.workingText")
+//healthCheckHandler is the handler responsible for validating that the app is still up
+func (app *App) healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	workingString := app.Config.GetString("healthcheck.workingText")
+	w.Header().Add("Content-Type", "text/plain; charset=utf-8")
+	res, err := app.Leaderboards.Ping(context.Background())
+	var msg string
+	if err != nil || res != "PONG" {
+		w.WriteHeader(http.StatusInternalServerError)
+		msg = newFailMsg(fmt.Sprintf("Error connecting to redis: %v", err))
+	} else {
+		msg = workingString
+	}
 
-		err := WithSegment("redis", c, func() error {
-			res, err := app.Leaderboards.Ping(c.StdContext())
-			if err != nil || res != "PONG" {
-				return fmt.Errorf("Error connecting to redis: %s", err)
-			}
+	if _, err := w.Write([]byte(msg)); err != nil {
+		app.Logger.Error("Error writing /healthcheck response", zap.Error(err))
+	}
+}
 
-			return nil
-		})
+func (app *App) HealthCheck(ctx context.Context, in *api.HealthCheckRequest) (*api.HealthCheckResponse, error) {
+	var res string
+
+	err := withSegment("redis", ctx, func() error {
+		var err error
+		res, err = app.Leaderboards.Ping(ctx)
 		if err != nil {
```

Comment:
Consider a switch instead of `else if` chains, it's idiomatic Go.

```golang
switch res, err = app.Leaderboards.Ping(ctx); {
case err != nil:
  // ...
case res != "PONG":
  // ...
}
```

Reply:
Awesome.